tests/test_utils.py

```diff
@@ -40,12 +41,30 @@ def test_broadcast_cshapes(cshape, reference):
         assert isinstance(broadcast, type(signal))
 
 
+@pytest.mark.parametrize("shape1, shape2, n_samples, caxis, ref1, ref2", [
```

Comment:
would suggest to rename the parameters to `cshape1` and `cshape2` to make it more clear. Is n_samples really needed as a variable? If not the test could always use the same number of samples and the variable could be removed.